### ISSUE TITLE
Remove entropy term from desc `softmax`

### ISSUE THREAD

**Issue by dpanici** [PlasmaControl/DESC#1317]
[Updated issue](https://github.com/PlasmaControl/DESC/issues/1317#issuecomment-2434352292)

I a while back implemented softmax [here](https://github.com/PlasmaControl/DESC/blob/d138a4990d7a1c4d50e0f9e781c43b5d24c16b02/desc/objectives/utils.py#L285)

but I realize we also have [since grabbed the one from `jax.nn`](https://github.com/PlasmaControl/DESC/blob/d138a4990d7a1c4d50e0f9e781c43b5d24c16b02/desc/backend.py#L74)

We should probably use just the one from `jax.nn` assuming they are interchangable

**Comment by dpanici** [comment]
so `jax.nn.softmax` returns an array
$exp(x_i)/\sumexp(x_i)$

while if we actually want the soft approx to maximum, what we want (and what our softmax implements) is
$softmax = log(\sum(exp(x_i))$

Technically I guess we could do something convoluted in the code to call the `jax.nn.softmax` to do the same thing but I think it is better to use jax's `logsumexp` implementation instead, and ends up being cleaner code. Closing this as unnecessary to do as they are really two different functions (hence Kaya's apt naming of the `jax.nn.softmax` as `softargmax` upon import to avoid confusion with the name

**Comment by unalmis** [comment]
@dpanici `softmax(x)` is typically implemented as
```python
from jax.nn import softmax as softargmax

softmax = x.dot(softargmax(x, beta=beta))
```

**Comment by unalmis** [comment]
The current implementation in DESC is not softmax.

Softmax(x) is the continuous approximation to hardmax(x). This is best formulated as the expected value of x under the softmax probability distribution, which can be computed with the code I have given above. On the other hand `logsumexp(x)` includes an additional entropy term.

$\text{logsumexp}(x) = \text{softmax}(x) + H(\text{softargmax(x)})$

This is not an ignorable term. Consider when x is a constant array with value $y$ (maximum entropy). Then softmax(x) = y. However, logsumexp(x) > y. When y =1, this is significant.

**Comment by dpanici** [comment]
Thanks for catching this, that's my bad
